src/naive_bayes/baseNaiveBayes.ts

```diff
@@ -0,0 +1,149 @@
+/**
+*  @license
+* Copyright 2022, JsData. All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* ==========================================================================
+*/
+import { polyfillUnique } from '../tfUtils'
+import { tf } from '../shared/globals'
+import { Scikit1D, Scikit2D } from '../types'
+import { convertToNumericTensor2D, convertToTensor1D } from '../utils'
+import { Tensor1D } from '@tensorflow/tfjs-core'
+
+export interface NaiveBayesParams {
+  /**
+   * Prior probabilities of the classes. If specified the priors are not adjusted according to the data.
+   */
+  priors?: Scikit1D
+
+  /**
+   * Portion of the largest variance of all features that is added to variances for calculation stability.
+   * Default value is 1e-9
+   */
+  varSmoothing?: number
+}
+
+export abstract class BaseNaiveBayes {
+  priors?: tf.Tensor1D
+  varSmoothing: number
+
+  private classes: tf.Tensor1D
+  private means: tf.Tensor1D[]
+  private variances: tf.Tensor1D[]
+
+  constructor(params: NaiveBayesParams) {
```

Comment:
`constructor(params: NaiveBayesParams = {})`

That way someone doesn't need to pass an empty object. If you don't have the default object then the following code errors

```
let nb = GaussianNB() // would error without default arg
```